daemon/openrazer_daemon/misc/complex_effect.py

```diff
@@ -0,0 +1,231 @@
+import threading
+import json
+import logging
+
+# pylint: disable=import-error
+from openrazer_daemon.keyboard import KeyboardColour
+from openrazer_daemon.misc.utils import capitalize_first_char
+
+def hex2rgb(hx):
+    if len(hx) != 7:
+        return (255, 0, 0)
+    if hx[0] != '#':
+        return (255, 0, 0)
+    try:
+        return (
+            int(hx[1:3], 16), int(hx[3:5], 16), int(hx[5:7], 16)
+        )
+    except:
+        return (255, 0, 0)
+def transmute_keyboard_matrix_from_polychromatic(raw_matrix):
```

Comment:
Not sure about naming the function after the app. What's it reading from Polychromatic?

Reply:
@lah7 It uses the layout of the keyboard matrix coloring object from polychromatic's config JSONs. I've made it this way because I found it super simple to design the colors in polychromatic, then simply copy the relevant part of the config file over.